pkg/pillar/hypervisor/kvm.go

```diff
@@ -426,7 +426,7 @@ const qemuSwtpmTemplate = `
   chardev = "swtpm"
 
 [device "tpm-tis"]
-  driver = "tpm-tis"
+  driver = "{{.TisDriver}}"
```

Comment:
@shjala , I think we don't need all of this.... would be the following sufficient?

```diff
diff --git a/pkg/pillar/hypervisor/kvm.go b/pkg/pillar/hypervisor/kvm.go
index e91af154e..d97f6847c 100644
--- a/pkg/pillar/hypervisor/kvm.go
+++ b/pkg/pillar/hypervisor/kvm.go
@@ -426,7 +426,11 @@ const qemuSwtpmTemplate = `
   chardev = "swtpm"
 
 [device "tpm-tis"]
+{{- if eq .Machine "virt"}}
+  driver = "tpm-tis-device"
+{{- else}}
   driver = "tpm-tis"
+{{- end}}
   tpmdev = "tpm0"
 `
```

Reply:
That's how it's done, for instance, for the serial passthrough, which uses different drivers for x86 and arm.....

Reply:
is "virt" a known machine type that corresponds to amr64?

Reply:
Just curious, why `{{- if eq .Machine "virt"}}` stands for arm ...
I think @shjala s approach is more explicit... But if we already do it this way... I don't know...

Reply:
Yes, that's the machine model used for arm...

Reply:
I see it is filled by devicemodel, but I like to be more clear.

Reply:
changed it to what @rene suggested, keeping the syntax consistent.

Reply:
> Just curious, why `{{- if eq .Machine "virt"}}` stands for arm ... I think @shjala s approach is more explicit... But if we already do it this way... I don't know...



> Just curious, why `{{- if eq .Machine "virt"}}` stands for arm ... I think @shjala s approach is more explicit... But if we already do it this way... I don't know...

It's not our fault, that's the name of the device model of QEMU: https://github.com/lf-edge/eve/blob/master/pkg/pillar/hypervisor/kvm.go#L473 there are many other ifs like this one in the templates...

Reply:
> changed it to what @rene suggested, keeping the syntax consistent.

Yeah, makes sense... I would hope the next time we add something under this check - we also add a comment on that, explaining what's happening, as @shjala did.

Reply:
> > Just curious, why `{{- if eq .Machine "virt"}}` stands for arm ... I think @shjala s approach is more explicit... But if we already do it this way... I don't know...
> 
> > Just curious, why `{{- if eq .Machine "virt"}}` stands for arm ... I think @shjala s approach is more explicit... But if we already do it this way... I don't know...
> 
> It's not our fault, that's the name of the device model of QEMU: https://github.com/lf-edge/eve/blob/master/pkg/pillar/hypervisor/kvm.go#L473 there are many other ifs like this one in the templates...

I see you also have a problem of doubling quoted messages =( Sometimes I even have three of them (

Reply:
@OhmSpectator TBH I think this was the first time.... or I didn't notice before, who knows..... 

Thanks @shjala , the comment in the template is really useful...